app/models/concerns/folio/has_site_roles.rb

```diff
@@ -5,38 +5,93 @@ module Folio::HasSiteRoles
 
   included do
     validate :validate_site_roles
+    validate :validate_site_link_uniqueness
 
     has_many :site_user_links, class_name: "Folio::SiteUserLink",
                                foreign_key: :user_id,
                                inverse_of: :user,
                                dependent: :destroy
     has_many :sites, through: :site_user_links,
                      source: :site
+
+    scope :with_unscoped_roles, -> (roles) { where(id: Folio::SiteUserLink.by_roles(roles).pluck(:user_id)) }
```

Comment:
@foton ne pluck, ale select - s `pluck` jsou to dva SQL prikazy a jeste zbytecne nacita velke pole do pameti